File: job.js
const fs = require('fs');
const path = require('path');

// 1. Update script.js
const scriptPath = 'script.js';
if (fs.existsSync(scriptPath)) {
    let c = fs.readFileSync(scriptPath, 'utf8');

    const target = `    // Distribuir para o contentor correto baseado no tipo
    if (expense.kind !== "fixed" && variableContainer) {
        variableContainer.classList.remove("empty-state");
    variableContainer.classList.add("item-list");
        variableContainer.appendChild(node);
    }`;

    const replacement = `    // Distribuir para o contentor geral
    if (variableContainer) {
        if (expense.kind === "fixed") {
            node.querySelector(".item-title").innerHTML += ' <span class="badge" style="font-size:0.65rem; opacity:0.9; margin-left:6px; background:rgba(13, 148, 136, 0.1); border: 1px solid rgba(13,148,136,0.3); color: var(--primary); padding:2px 6px; border-radius:4px;">Fixa</span>';
        }
        variableContainer.classList.remove("empty-state");
        variableContainer.classList.add("item-list");
        variableContainer.appendChild(node);
    }`;

    c = c.replace(target, replacement);
    fs.writeFileSync(scriptPath, c);
}

// 2. Update HTML files
const files = ['dashboard.html', 'extrato.html', 'configuracao.html', 'index.html'];
files.forEach(file => {
    if (fs.existsSync(file)) {
        let hc = fs.readFileSync(file, 'utf8');
        
        hc = hc.replace(/class="([^"]*item-list[^"]*)"/g, (match, cls) => {
            if (!cls.includes('item-list-container') && cls.includes('item-list')) {
                return `class="${cls} item-list-container"`;
            }
            return match;
        });
        
        fs.writeFileSync(file, hc);
    }
});

console.log('Success');

File: job.py
import os, re

# 1. Update script.js
path = 'script.js'
with open(path, 'r', encoding='utf-8') as f:
    c = f.read()

# Make fixed expenses show in the general expenses container
target = '''    // Distribuir para o contentor correto baseado no tipo
    if (expense.kind !== "fixed" && variableContainer) {
        variableContainer.classList.remove("empty-state");
    variableContainer.classList.add("item-list");
        variableContainer.appendChild(node);
    }'''

replacement = '''    // Distribuir para o contentor geral
    if (variableContainer) {
        if (expense.kind === "fixed") {
            node.querySelector(".item-title").innerHTML += ' <span class="badge badge-primary" style="font-size:0.65rem; opacity:0.8; margin-left:6px; background:rgba(13, 148, 136, 0.1); color: var(--primary); padding:2px 6px; border-radius:4px;">Fixa</span>';
        }
        variableContainer.classList.remove("empty-state");
        variableContainer.classList.add("item-list");
        variableContainer.appendChild(node);
    }'''

c = c.replace(target, replacement)
with open(path, 'w', encoding='utf-8') as f:
    f.write(c)

# 2. Update HTML files to add item-list-container to all item-lists
for file in ['dashboard.html', 'extrato.html', 'configuracao.html', 'index.html']:
    if os.path.exists(file):
        with open(file, 'r', encoding='utf-8') as f:
            hc = f.read()
            
        def replacer(match):
            cls = match.group(1)
            if 'item-list-container' not in cls and 'item-list' in cls:
                cls += ' item-list-container'
            return 'class="' + cls + '"'
            
        hc = re.sub(r'class="([^"]*item-list[^"]*)"', replacer, hc)
        with open(file, 'w', encoding='utf-8') as f:
            f.write(hc)

print('Success')
